includes/html/graphs/generic_multi_seperated.inc.php

```diff
@@ -19,9 +19,19 @@
 
 $rrddescr_len = 14; // length of the padded rrd_descr in legend
 
+if ($width > '1500') {
+    $rrddescr_len = 30;
+} elseif ($width >= '500') {
+    $rrddescr_len = 8;
+    $rrddescr_len += min(30, round(($width - 320) / 15));
+} else {
+    $rrddescr_len = 8;
+    $rrddescr_len += min(20, round(($width - 260) / 9.5));
+}
+
 $stacked = generate_stacked_graphs();
 
-$units_descr = substr(str_pad($units_descr, $rrddescr_len + 9), 0, $rrddescr_len + 9);
+$units_descr = substr(rrdtool_escape($units_descr, $rrddescr_len + 5), 0, $rrddescr_len + 5);
```

Comment:
substr is not needed

Reply:
not sure why it was there originally. but you're right.